server/src/main/java/org/elasticsearch/action/support/IndicesOptions.java

```diff
@@ -1099,23 +1100,27 @@ public static IndicesOptions fromOptions(
         );
     }
 
-    public static IndicesOptions fromRequest(RestRequest request, IndicesOptions defaultSettings) {
-        if (request.hasParam(GatekeeperOptions.IGNORE_THROTTLED)) {
+    public static IndicesOptions fromRequestParams(RequestParams params, IndicesOptions defaultSettings) {
```

Comment:
Added a method to get IndicesOptions from RequestParams, the existing one delegates to this